src/community/features-templating/features-templating-ows/src/main/java/org/geoserver/featurestemplating/ows/TemplateCallback.java

```diff
@@ -159,18 +160,38 @@ private void replaceTemplatePath(Query q, FeatureTypeInfo fti, RootBuilder root)
     public Response responseDispatched(
             Request request, Operation operation, Object result, Response response) {
         Object[] params = operation.getParameters();
-        if (params.length > 0) {
+        if (operationSupported(operation) && params.length > 0) {
             Object param1 = params[0];
             Response replacer = findResponse(param1);
             if (replacer != null) response = replacer;
         }
         return super.responseDispatched(request, operation, result, response);
     }
 
+    /**
+     * Checks if operation is supported by features templating
+     *
+     * @param operation
+     * @return
+     */
+    private boolean operationSupported(Operation operation) {
+        String id = operation.getId();
+        String serviceId = operation.getService().getId();
+        return id.equals("GetFeatureInfo") && serviceId.equalsIgnoreCase("wms")
+                || id.equals("GetFeature") && serviceId.equalsIgnoreCase("wfs");
+    }
+
     private Response findResponse(Object param1) {
         Response response = null;
         if (param1 instanceof GetFeatureInfoRequest) {
-            response = getTemplateFeatureInfoResponse((GetFeatureInfoRequest) param1);
+            GetFeatureInfoRequest request = (GetFeatureInfoRequest) param1;
+            // Checking that template format is specified and that it's not text/plain because it's
+            // not supported,
+            // but is default if nothing is specified
+            if (request.getInfoFormat() != null
+                    && !request.getInfoFormat().equals(MediaType.TEXT_PLAIN_VALUE)) {
+                response = getTemplateFeatureInfoResponse(request);
+            }
```

Comment:
Output formats are pluggable, meaning others not supported could be added tomorrow and this exclusion code would stop working. Can we have a white list of supported formats instead? Do you think it would work better, or would it also break?

Reply:
I think that best solution would be something dynamic, similar to GeoServerExtensions.extensions() to return all available formats and compare against them. But I need to check it.